### DISCUSSION TITLE
Access tracking matrix from BrainLab en sending it to a python script

### DISCUSSION THREAD

**Discussion by ReiniertB** [PlusToolkit/PlusLib#981] [Q&A]
Hi,

I am trying to access the tracking data from an instrument in BrainLab as a matrix and sending that to a python script.
I first tried this without using the PlusToolkit using the following tutorial:
[https://www.slicer.org/wiki/Documentation/4.10/Modules/OpenIGTLinkRemote/BrainlabTutorial](url)
My plan was to have a client in Slicer receive the tracking data from BrainLab, and then have a device with the same name and different port as a server. Then in my python script, using pyitl I should be able to receive these coordinates as a matrix.
Unfortunately, while I have access to IGT Link in BrainLab and also was able to push imaging data to Slicer, I was not able to receive tracking data.
I thought I perhaps needed the PlusToolKit, but I am a bit lost in the woods with all the config files and everything that is in there.
I tried to alter the config file from this link [http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceOpenIGTLinkTracker.html](url) with the IP adress from my BrainLab system etc.
I have the feeling my config file should be a lot easier than this but I don't know where to start and what to do with all those transformations. Could someone please help me? Thanks in advance

**Comment by lassoan** [comment]
BrainLab uses TDATA message to send transforms. Currently pyigtl only supports TRANSFORM message. Your can use PlusServer to connect to BrainLab to receive TDATA message and broadcast them as TRANSFORM messages. Alternatively, you can import TDATA message support in pyigtl.

**Reply by ReiniertB** [reply]
Hi, 

I have been able to receive tracking data from BrainLab without using PLUS by using an older version of Slicer.
However, I want to access the transformation matrix from the Pointer. So I made a config file which would broadcast this data and send it to slicer unaltered. But I get the following error 
![image](https://user-images.githubusercontent.com/75205068/191738986-d88e75d9-dce0-4030-b81b-e52eaea7b5fa.png)

This is my config file (please don't mind the server adress, this is currently a local host which is a python file which just sends out a bunch of 1's so I don't have to use BrainLab all the time) (and I know its an txt file, but I can't upload xml files): 
Thank you very much in advance!

[PlusDeviceSet_Server_BrainLab_test.txt](https://github.com/PlusToolkit/PlusLib/files/9625370/PlusDeviceSet_Server_BrainLab_test.txt)
